### ISSUE TITLE
Logic error, page 573

### ISSUE THREAD

**Issue by chessboards** [head-first-csharp/fourth-edition#24]
`if (i < bytesRead)` fails to handle the last byte of those in the buffer. should be changed to `if (i <= bytesRead)`.

also making it a bit clearer that there are 3 spaces for formatting would improve readability

**Comment by andrewstellman** [comment]
Good catch, I think you're right. We'll definitely make sure this works in future printings, especially with the upcoming VS2022.
